### DISCUSSION TITLE
Mock data from BatchLoader

### DISCUSSION THREAD

**Discussion by IvanDanilau** [Netflix/dgs-framework#571] [Q&A]
I 've mock  some data, but when I add field which is loaded with BatchLoader, it's still called. How can I do the mocking? Perhaps some special key value?

**Comment by paulbakker** [comment]
You'll have to elaborate a bit more about what you're trying.

**Reply by IvanDanilau** [reply]
I have DataFether part:

 @DgsQuery(field = QUERY.FindUser)
  public Optional<UserDTO> findUser(@InputArgument Long id) {
    return userService.findOne(id);
  }

  @DgsData(parentType = USER.TYPE_NAME, field = USER.Contacts)
  public CompletableFuture<UserContactInfoDTO> findUserContacts(DgsDataFetchingEnvironment dfe) {
    UserDTO user = dfe.getSource();
    DataLoader<String, UserContactInfoDTO> dataLoader = dfe
        .getDataLoader(UserContactsDataLoader.class);
    return dataLoader.load(user.getId());
  }

I want use mock data for this API according to the documentation:

@Component
@RequiredArgsConstructor(onConstructor_ = {@Autowired})
public class UserMockProvider implements MockProvider {

  private final GraphMockDataStorage dataStorage;

  @NotNull
  @Override
  public Map<String, Object> provide() {
    Map<String, Object> mock = new HashMap<>();
    UserDTO userMock = dataStorage.userMock();
    mock.put(QUERY.FindUser, userMock);
    mock.put(USER.Contacts, dataStorage.userContactsMock());
    return mock;
  }


So the issue is that method from UserContactsDataLoader still executed. But I expect getting mock data, the same as for user info(for user it works fine).
